### PR TITLE
Redux State to Components

### PR DESCRIPTION
**Pull request by wwwdrt**
<h1 align="center">Hey CRs</h1>

<div align="center">
Please feel free to reach out to me on Slack anytime. You can send me a direct message or give me a call. If you choose to call, please introduce yourself so I know who I'm speaking with. I'm here to help, so don't hesitate to get in touch 😁
</div>

<br>

<div align="center">
  

### Latest project 🔥"[AI Quiz Generation Platform](https://github.com/wwwdrt/quizzen)"
</div>

<h1 align="center">Changes</h1>

<div align="center">
In this branch, I have passed redux state to react components.
</div>

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
+  } = book;
 
-const Book = ({ title, author }) => {
   const [isDeleted, setIsDeleted] = useState(false);
+  const dispatch = useDispatch();
 
   const handleDelete = () => {
     setIsDeleted(true);
+    dispatch(removeBook(id));
   };
```
Comment (Efiamotu-1) [inline src/components/Book.js:19]: - [ ] Currently when i add multiple books for example I add two books and I try to remove one of the books I added, the two previously added books gets removed, this shouldn't be the functionality of the application, please kindly work on the handleDelete to delete the exact book specified 👍

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @wwwdrt, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Book.js (hunk 1 above)
Comment (wwwdrt) [reply src/components/Book.js:19]: Screenshot at 2023-08-22 18-16-22

:heavy_check_mark:

**Review by cynthiainga — APPROVED** [review]
Hi @wwwdrt, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] User can add and delete a book ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**